ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java

```diff
@@ -50,6 +51,7 @@ protected ReactActivityDelegate createReactActivityDelegate() {
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     mDelegate.onCreate(savedInstanceState);
+    OkHttpClientProvider.addNetworkListenerToEvictClientConnectionsOnDisconnect(getApplicationContext());
```

Comment:
This name strikes me as very Java. :stuck_out_tongue:

(Which is not a complaint! We are after all writing Java here. The name is good and clear.)

Reply:
Rereading this name, I do actually have a suggestion, though: s/ClientConnections/IdleConnections/. That more closely reflects what that `evictAll` method does (according to OkHttp's docs), and might avoid someone worrying that this cure could be causing its own disease by closing active connections that might have successfully finished their requests with a bit of patience.

Reply:
Heh, though now that you've added the `cancelAll`, this new name isn't accurate -- in fact it sounds too innocent, which is dangerous.

Maybe s/EvictIdleConnections/DropConnections/g? It's a bit more vague, leaving room for the specifics of what goes inside that `sClients` for-loop to vary. And appropriately scary. :-p